src/Components/Web.JS/src/Boot.Server.Common.ts

```diff
@@ -65,6 +65,10 @@ export async function startServer(components: RootComponentManager<ServerCompone
   Blazor._internal.sendJSDataStream = (data: ArrayBufferView | Blob, streamId: number, chunkSize: number) => circuit.sendJsDataStream(data, streamId, chunkSize);
 
   const jsInitializer = await fetchAndInvokeInitializers(options);
+
+  appState = discoverServerPersistedState(document) || '';
```

Comment:
There is an existing bug in persisting state. Circuit starts without restoring the app state and the comment containing the state is still on the page. The reason is because circuit is created before the comment containing app is emitted to the page. Solution is to set the app state later

Reply:
@surayya-MS I'm not sure that's right.

@MackinnonBuck shouldn't we delay starting the circuit until the state is available?

To be very clear, the state needs to be provided alongside the components at the time of initializing the circuit in an atomic fashion (components can't start rendering until their state is available).